lib/net.js

```diff
@@ -299,6 +302,10 @@ Socket.prototype.listen = function() {
 
 
 Socket.prototype.setTimeout = function(msecs, callback) {
+  if (typeof msecs === 'function') {
```

Comment:
This is here to trap for when a user does `socket.setTimeout.promise().then()`, since msecs will be omitted, but callback will be present — an error state the API did not track previously. This also brings `socket.setTimeout` in line with global `setTimeout`.